File: SoftInvModel/src/main/java/pe/edu/pucp/softinv/model/Producto/ProductoDTO.java
package pe.edu.pucp.softinv.model.Producto;

import pe.edu.pucp.softinv.model.Comentario.ComentarioProductoDTO;

import java.util.ArrayList;

public class ProductoDTO {
    private Integer idProducto;
    private String nombre;
    private String descripcion;
    private Double precio;
    private ArrayList<String> ingredientes;
    private String modoUso;
    private TipoProducto tipoProducto;
    private ArrayList<String>tamanios;
    private ArrayList<TipoPiel> tipoPiel;
    private Integer stock;
    private String urlImagen;
    private ArrayList<ComentarioProductoDTO> comentarios;
    private Double promedioValoracion;

    public ArrayList<String> getTamanios() {
        return tamanios;
    }

    public void setTamanios(ArrayList<String> tamanios) {
        this.tamanios = tamanios;
    }

    public Integer getIdProducto() {
        return idProducto;
    }

    public void setIdProducto(Integer idProducto) {
        this.idProducto = idProducto;
    }

    public Double getPromedioValoracion() {
        return promedioValoracion;
    }

    public void setPromedioValoracion(Double promedioValoracion) {
        this.promedioValoracion = promedioValoracion;
    }

    public ArrayList<ComentarioProductoDTO> getComentarios() {
        return comentarios;
    }

    public void setComentarios(ArrayList<ComentarioProductoDTO> comentarios) {
        this.comentarios = comentarios;
    }

    public String getUrlImagen() {
        return urlImagen;
    }

    public void setUrlImagen(String urlImagen) {
        this.urlImagen = urlImagen;
    }

    public Integer getStock() {
        return stock;
    }

    public void setStock(Integer stock) {
        this.stock = stock;
    }

    public ArrayList<TipoPiel> getTipoPiel() {
        return tipoPiel;
    }

    public void setTipoPiel(ArrayList<TipoPiel> tipoPiel) {
        this.tipoPiel = tipoPiel;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getDescripcion() {
        return descripcion;
    }

    public void setDescripcion(String descripcion) {
        this.descripcion = descripcion;
    }

    public Double getPrecio() {
        return precio;
    }

    public void setPrecio(Double precio) {
        this.precio = precio;
    }

    public ArrayList<String> getIngredientes() {
        return ingredientes;
    }

    public void setIngredientes(ArrayList<String> ingredientes) {
        this.ingredientes = ingredientes;
    }

    public String getModoUso() {
        return modoUso;
    }

    public void setModoUso(String modoUso) {
        this.modoUso = modoUso;
    }

    public TipoProducto getTipoProducto() {
        return tipoProducto;
    }

    public void setTipoProducto(TipoProducto tipoProducto) {
        this.tipoProducto = tipoProducto;
    }

    public ProductoDTO() {
        idProducto=null;
        tipoProducto = null;
        tipoPiel = null;
        nombre = null;
        descripcion = null;
        precio = null;
        ingredientes = null;
        modoUso = null;
        stock=null;
        urlImagen=null;
        comentarios = null;
        promedioValoracion=null;
        tamanios=null;
    }

    public ProductoDTO(Integer idProducto,String nombre,String descripcion,Double precio,
                       ArrayList<ComentarioProductoDTO> comentarios,ArrayList<String> ingredientes,String modoUso,
                       TipoProducto tipoProducto,ArrayList<String> tamanios,ArrayList<TipoPiel> tipoPiel,
                       Integer stock,String urlImagen,Double promedioValoracion) {
        this.idProducto=idProducto;
        this.nombre = nombre;
        this.descripcion = descripcion;
        this.precio = precio;
        this.comentarios = comentarios;
        this.ingredientes = ingredientes;
        this.modoUso = modoUso;
        this.tipoProducto = tipoProducto;
        this.tamanios = tamanios;
        this.tipoPiel = tipoPiel;
        this.stock = stock;
        this.urlImagen = urlImagen;
        this.promedioValoracion = promedioValoracion;
    }
}

File: C#/SoftInvModel/Producto/ProductoDTO.cs
using System;
using System.ComponentModel;

namespace softinv.model
{
    public class ProductoDTO
    {
        private int? idProducto;
        private string nombre;
        private string descripcion;
        private double? precio;
        private BindingList<string> ingredientes;
        private string modoUso;
        private TipoProducto? tipoProducto;
        private BindingList<TipoPiel> tipoPiel;
        private int? stock;
        private string urlImagen;
        private BindingList<ComentarioProductoDTO> comentarios;
        private double? promedioValoracion;

        public ProductoDTO()
        {
            IdProducto = null;
            TipoProducto = null;
            TipoPiel = null;
            Nombre = null;
            Descripcion = null;
            Precio = null;
            Ingredientes = null;
            ModoUso = null;
            Stock = null;
            UrlImagen = null;
            Comentarios = null;
            PromedioValoracion = null;
        }

        public ProductoDTO(int idProducto, string nombre, string descripcion, Double precio,
                           BindingList<ComentarioProductoDTO> comentarios, BindingList<string> ingredientes, string modoUso,
                           TipoProducto tipoProducto, BindingList<string> tamanios, BindingList<TipoPiel> tipoPiel,
                           int stock, string urlImagen, Double promedioValoracion)
        {
            this.IdProducto = idProducto;
            this.Nombre = nombre;
            this.Descripcion = descripcion;
            this.Precio = precio;
            this.Comentarios = comentarios;
            this.Ingredientes = ingredientes;
            this.ModoUso = modoUso;
            this.TipoProducto = tipoProducto;
            this.TipoPiel = tipoPiel;
            this.Stock = stock;
            this.UrlImagen = urlImagen;
            this.PromedioValoracion = promedioValoracion;
        }

        public int? IdProducto { get => idProducto; set => idProducto = value; }
        public string Nombre { get => nombre; set => nombre = value; }
        public string Descripcion { get => descripcion; set => descripcion = value; }
        public double? Precio { get => precio; set => precio = value; }
        public BindingList<string> Ingredientes { get => ingredientes; set => ingredientes = value; }
        public string ModoUso { get => modoUso; set => modoUso = value; }
        public TipoProducto? TipoProducto { get => tipoProducto; set => tipoProducto = value; }
        public BindingList<TipoPiel> TipoPiel { get => tipoPiel; set => tipoPiel = value; }
        public int? Stock { get => stock; set => stock = value; }
        public string UrlImagen { get => urlImagen; set => urlImagen = value; }
        public BindingList<ComentarioProductoDTO> Comentarios { get => comentarios; set => comentarios = value; }
        public double? PromedioValoracion { get => promedioValoracion; set => promedioValoracion = value; }
    }

}
